plugins/module_utils/type.py

```diff
@@ -8,11 +8,25 @@
 
 __metaclass__ = type
 
-import enum
+from typing import Optional
+
+FROM_HYPERCORE_TO_ANSIBLE_NIC_TYPE = {
+    None: None,
+    "RTL8139": "RTL8139",
+    "VIRTIO": "virtio",
+    "INTEL_E1000": "INTEL_E1000",
+}
+FROM_ANSIBLE_TO_HYPERCORE_NIC_TYPE = {
+    v: k for k, v in FROM_HYPERCORE_TO_ANSIBLE_NIC_TYPE.items()
+}
 
 
 # Maybe create enums.py or scale_enums.py and move all enum classes there? @Jure @Justin
-class NicType(str, enum.Enum):
-    RTL8139 = "RTL8139"
-    VIRTIO = "VIRTIO"
-    INTEL_E1000 = "INTEL_E1000"
+class NicType:
```

Comment:
I would just move this class to module_utils/nic.py (and delete type.py). But this is just my opinion, not necessary to change that. Otherwise nice job, I approve :)